File: lab2.java
import java.util.*;

class Meeting {
    int start, end, index;
    Meeting(int start, int end, int index) {
        this.start = start;
        this.end = end;
        this.index = index;
    }
}

public class lab2 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        System.out.print("Введите количество заявок: ");
        int N = sc.nextInt();
        
        List<Meeting> meetings = new ArrayList<>();
        for (int i = 0; i < N; i++) {
            int start = sc.nextInt();
            int end = sc.nextInt();
            meetings.add(new Meeting(start, end, i + 1));
        }
        
        meetings.sort((a, b) -> {
            if (a.end == b.end) return a.start - b.start;
            return a.end - b.end;
        });
        
        List<Integer> selectedIndices = new ArrayList<>();
        int lastEnd = -1;
        
        for (Meeting m : meetings) {
            if (m.start >= lastEnd) {
                selectedIndices.add(m.index);
                lastEnd = m.end;
            }
        }
        
        Collections.sort(selectedIndices);
        
        System.out.println("Результат: " + selectedIndices.size());
        for (int idx : selectedIndices) {
            System.out.print(idx + " ");
        }
        System.out.println();
        
        sc.close();
    }
}

File: lab2.swift
import Foundation

struct Meeting {
    let start: Int
    let end: Int
    let index: Int
}

print("Введите количество заявок:", terminator: " ")
guard let N = Int(readLine() ?? "") else { fatalError() }

var meetings: [Meeting] = []

for i in 1...N {
    let input = readLine() ?? ""
    let parts = input.split(separator: " ").map { Int($0)! }
    meetings.append(Meeting(start: parts[0], end: parts[1], index: i))
}

meetings.sort { 
    if $0.end == $1.end {
        return $0.start < $1.start
    }
    return $0.end < $1.end
}

var selectedIndices: [Int] = []
var lastEnd = -1

for m in meetings {
    if m.start >= lastEnd {
        selectedIndices.append(m.index)
        lastEnd = m.end
    }
}

selectedIndices.sort()

print("Результат:", selectedIndices.count)
print(selectedIndices.map { String($0) }.joined(separator: " "))